ZuluContent/Mobiles/BaseCreature/BaseCreature.cs

```diff
@@ -555,67 +555,52 @@ public virtual bool IsEnemy(Mobile m)
 
         public virtual bool CheckControlChance(Mobile m)
         {
-            if (GetControlChance(m) > Utility.RandomDouble())
-            {
-                Loyalty += 1;
+            if (CanBeControlledBy(m))
                 return true;
-            }
 
-            PlaySound(GetAngerSound());
-
-            if (Body.IsAnimal)
-                Animate(10, 5, 1, true, false, 0);
-            else if (Body.IsMonster)
-                Animate(18, 5, 1, true, false, 0);
+            AIObject.DoOrderRelease(true);
 
-            Loyalty -= 3;
             return false;
         }
 
         public virtual bool CanBeControlledBy(Mobile m)
         {
-            return GetControlChance(m) > 0.0;
-        }
-
-        public double GetControlChance(Mobile m)
-        {
-            return GetControlChance(m, false);
-        }
-
-        public virtual double GetControlChance(Mobile m, bool useBaseSkill)
-        {
-            if (m_dMinTameSkill <= 29.1 || m_bSummoned || m.AccessLevel >= AccessLevel.GameMaster)
-                return 1.0;
-
-            double dMinTameSkill = m_dMinTameSkill;
-
-            int taming =
-                (int) ((useBaseSkill ? m.Skills[SkillName.AnimalTaming].Base : m.Skills[SkillName.AnimalTaming].Value) *
-                       10);
-            int lore =
-                (int) ((useBaseSkill ? m.Skills[SkillName.AnimalLore].Base : m.Skills[SkillName.AnimalLore].Value) *
-                       10);
-            int bonus = 0, chance = 700;
-
-            int difficulty = (int) (dMinTameSkill * 10);
-            int weighted = (taming * 4 + lore) / 5;
-            bonus = weighted - difficulty;
+            if (m is PlayerMobile owner)
+            {
+                const int minSlots = 2;
+                int maxSlots;
+                var petSlots = 0;
 
-            if (bonus <= 0)
-                bonus *= 14;
-            else
-                bonus *= 6;
+                IPooledEnumerable eable = owner.Map.GetMobilesInRange(owner.Location, 8);
 
-            chance += bonus;
+                foreach (Mobile nearbyMobile in eable)
+                {
+                    if (nearbyMobile is BaseCreature creature &&
+                        (creature.ControlMaster == owner || creature.SummonMaster == owner))
+                    {
+                        if (creature.Summoned && creature.Serial != Serial &&
+                            creature.IsElementalLord)
```

Comment:
I think we can keep the `IsElementalLord` inlined, unless there's other code we haven't implemented yet that uses it?

It should also be a type-check against the creature types instead checking the string name.